kratos/solving_strategies/strategies/line_search_strategy.h

```diff
@@ -147,7 +151,44 @@ class LineSearchStrategy
         bool ReformDofSetAtEachStep = false,
         bool MoveMeshFlag = false
     ): ResidualBasedNewtonRaphsonStrategy<TSparseSpace, TDenseSpace, TLinearSolver>(model_part, pScheme, pNewLinearSolver,pNewConvergenceCriteria,pNewBuilderAndSolver,MaxIterations,CalculateReactions,ReformDofSetAtEachStep, MoveMeshFlag)
-    {}
+    {
+        Parameters default_settings = GetDefaultSettings();
+        this->mSettings.AddMissingParameters(default_settings);
+        AssignSettings();
+    }
+
+    LineSearchStrategy(
+        ModelPart& model_part,
+        typename TSchemeType::Pointer pScheme,
+        typename TLinearSolver::Pointer pNewLinearSolver,
+        typename TConvergenceCriteriaType::Pointer pNewConvergenceCriteria,
+        Parameters Settings,
+        bool CalculateReactions = false,
+        bool ReformDofSetAtEachStep = false,
+        bool MoveMeshFlag = false
+    ): ResidualBasedNewtonRaphsonStrategy<TSparseSpace, TDenseSpace, TLinearSolver>(model_part, pScheme, pNewLinearSolver,pNewConvergenceCriteria,Settings,CalculateReactions,ReformDofSetAtEachStep, MoveMeshFlag)
+    {
+        Parameters default_settings = GetDefaultSettings();
+        this->mSettings.AddMissingParameters(default_settings);
+        AssignSettings();
+    }
+
+    LineSearchStrategy(
+        ModelPart& model_part,
+        typename TSchemeType::Pointer pScheme,
+        typename TLinearSolver::Pointer pNewLinearSolver,
+        typename TConvergenceCriteriaType::Pointer pNewConvergenceCriteria,
+        typename TBuilderAndSolverType::Pointer pNewBuilderAndSolver,
+        Parameters Settings,
+        bool CalculateReactions = false,
+        bool ReformDofSetAtEachStep = false,
+        bool MoveMeshFlag = false
+    ): ResidualBasedNewtonRaphsonStrategy<TSparseSpace, TDenseSpace, TLinearSolver>(model_part, pScheme, pNewLinearSolver,pNewConvergenceCriteria,pNewBuilderAndSolver,Settings,CalculateReactions,ReformDofSetAtEachStep, MoveMeshFlag)
+    {
+        Parameters default_settings = GetDefaultSettings();
+        this->mSettings.AddMissingParameters(default_settings);
```

Comment:
why do you store the settings as a member?  you only need it in the constructor, socyou shoukd actually act on the one that are oassed to you. 

this way your modificstions will be visible outside the class

Reply:
Ok, I actually did not think about it. I just did it as it is usually done in the core but you are right, there is no really good reason to store it.